grpc-interface-support/src/main.py

```diff
@@ -15,24 +15,92 @@
 import argparse
 import os
 import shutil
+import zipfile
 from pathlib import Path
-from typing import Any, Dict
+from typing import Any, Dict, List
+from urllib.parse import urlparse
 
 import proto
 from cpp import CppGrpcServiceSdkGeneratorFactory
 from generator import GrpcServiceSdkGeneratorFactory
 from python import PythonGrpcServiceSdkGeneratorFactory
 from velocitas_lib import (
+    download_file,
     get_programming_language,
     get_project_cache_dir,
     obtain_local_file_path,
+    get_workspace_dir
 )
 from velocitas_lib.functional_interface import get_interfaces_for_type
 from velocitas_lib.text_utils import create_truncated_string
 
 DEPENDENCY_TYPE_KEY = "grpc-interface"
 
 
+def extract_zip(file_path: str, extract_to: str) -> str:
+    """Extrcat a zip file.
+
+    Args:
+        file_path (str): The file path to the zip.
+        extract_to (str): The fiel path to extract to.
+
+    Returns:
+        str: The file path to the extracted folder.
+    """
+    with zipfile.ZipFile(file_path, "r") as zip_ref:
+        zip_ref.extractall(extract_to)
+        return extract_to
+
+
+def find_proto_files(directory: str) -> List[proto.ProtoFileHandle]:
+    """
+    Recursively searches for .proto files under the specified directory.
+
+    Args:
+        directory (str): The path to the directory to search in.
+
+    Returns:
+        List[proto.ProtoFileHandle]: A list of proto files.
+    """
+    proto_files = []
+    for root, _, files in os.walk(directory):
+        for file in files:
+            if file.endswith(".proto"):
+                proto_files.append(proto.ProtoFileHandle(os.path.join(root, file)))
+    return proto_files
+
+
+def fetch_protos(path: str) -> List[proto.ProtoFileHandle]:
+    """Fetch the proto files defined in the grpc-interface
```

Comment:
Naming of `find_proto_files` vs `fetch_protos` is really confusing. Can we find better naming here or maybe merge the functions?

Reply:
would `obtain_protos` and `discover_protos` be better? Or do you have some suggestions yourself? Making it one function is a bit difficult.